lib/systems/include/TorqueControllerMux.tpp

```diff
@@ -0,0 +1,207 @@
+#include "TorqueControllerMux.h"
+
+template <std::size_t num_controllers>
+DrivetrainCommand_s TorqueControllerMux<num_controllers>::getDrivetrainCommand(ControllerMode_e requested_controller_type,
+                                                                               TorqueLimit_e requested_torque_limit,
+                                                                               const SharedCarState_s &input_state)
+{
+
+    DrivetrainCommand_s empty_command = BaseControllerParams::TC_COMMAND_NO_TORQUE;
+
+    TorqueControllerOutput_s current_output = {
+        .command = empty_command,
+        .ready = true};
+
+    int req_controller_mode_index = static_cast<int>(requested_controller_type);
+    int current_controller_mode_index = static_cast<int>(current_status_.current_controller_mode_);
+
+    if ((std::size_t)req_controller_mode_index > ( controller_pointers_.size() - 1 ))
+    {
+        current_status_.current_error = TorqueControllerMuxError::ERROR_CONTROLLER_INDEX_OUT_OF_BOUNDS;
+        return empty_command;
+    }
+
+    current_output = controller_pointers_[current_controller_mode_index]->evaluate(input_state);
```

Comment:
add a null pointer check?

Reply:
good point, adding test and check for this